cse/CommentReader.py

```diff
@@ -76,15 +114,26 @@ def __parseRow(self, row):
 
 
     def __parseIterRow(self, row):
+        if not self.__authors:
+            self.__loadAuthors()
+
         commentId = row[0]
-        articleUrl = row[1]
-        author = row[2]
+        articleId = row[1]
+        author = self.__authors[row[2]]
         text = row[3].replace("\\n", "\n")
         timestamp = row[4]
         parentId = row[5]
         upvotes = int(row[6])
         downvotes = int(row[7])
-        articleId = row[8]
+
+        if not self.__currentArticle:
+            self.__articlesFile.seek(0)
+            articlesFormat = next(self.__articlesReader)
+            self.__currentArticle = next(self.__articlesReader)
+
+        while self.__currentArticle[0] is not articleId:
+            self.__currentArticle = next(self.__articlesReader)
+        articleUrl = self.__currentArticle[1]
```

Comment:
you are reading (nearly) the whole article mapping file per line in the comments file. Is this intended? File reads are very expensive.

Currently we have no other way than iterating over the whole comments file per search query, which means we are reading the article file `number of lines in comments file` times

Reply:
since the files are ordered it will be exactly 1 read operation most of the time. Sometimes articles don't have comments. then we need more than 1 read operation. 

In the end the article maping file will be read exactly once.

Reply:
ahh I see, sry.
Well done!